### ISSUE TITLE
Unity 2018.3 appears to break rider integration on macOS

### ISSUE THREAD

**Issue by shadowmint** [JetBrains/resharper-unity#965]
Versions:

Rider: 2018.2.3
Unity: 2018.3.0f2
Platform: macOS Mojave 10.14.2

I have a working solution, but after upgrading to unity 2018.3 the rider integration appears to be broken.

The unity console shows no errors, and runs fine, but when opening the solution in rider and attempting a build, it looks like this:

```
Assembly-CSharp
  BindException.cs(7, 15): [CS0518] Assets/pkg-all/n-bind/BindException.cs(7,15): error CS0518: Predefined type 'System.Enum' is not defined or imported
  BindException.cs(7, 15): [CS0518] Assets/pkg-all/n-bind/BindException.cs(7,15): error CS0518: Predefined type 'System.Int32' is not defined or imported
  BindException.cs(23, 32): [CS0246] Assets/pkg-all/n-bind/BindException.cs(23,32): error CS0246: 
  ServiceRegistry.cs(13, 22): [CS0518] Assets/pkg-all/n-bind/ServiceRegistry.cs(13,22): error ect' is not defined or imported
  PropertyBinder.cs(17, 5): [CS0518] Assets/pkg-all/n-bind/core/PropertyBinder.cs(17,5): error CS0518: Predefined type 'System.Void' is not defined or imported
  Source.cs(3, 16): [CS0518] Assets/pkg-wip/N/Package/Cards/Source.cs(3,16): error CS0518: Predefined type 'System.Object' is not defined or imported
  .NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2, 46): [CS0234] obj/Debug/.NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2,46): error CS0234: The type or namespace name 'TargetFrameworkAttribute' does not exist in the namespace 'System.Runtime.Versioning' (are you missing an assembly reference?)
  .NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2, 46): [CS0234] obj/Debug/.NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2,46): error CS0234: The type or namespace name 'TargetFrameworkAttributeAttribute' does not exist in the namespace 'System.Runtime.Versioning' (are you missing an assembly reference?)
  .NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2, 71): [CS0518] obj/Debug/.NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2,71): error CS0518: Predefined type 'System.String' is not defined or imported
  .NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2, 103): [CS0246] obj/Debug/.NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2,103): error CS0246: The type or namespace name 'FrameworkDisplayName' could not be found (are you missing a using directive or an assembly reference?)
  .NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2, 126): [CS0518] obj/Debug/.NETFramework,Version=v4.7.1.AssemblyAttribute.cs(2,126): error CS0518: Predefined type 'System.String' is not defined or imported
```

^ 100s of other errors omitted, you get the idea. Anything in `System.*` is failing to resolve.

Rider's intellisense is also totally confused, for example it highlights 'bool' as an invalid type.

I've tried restarting, invalidating caches, etc.

This project worked fine before; I guess this is something new in 2018.3, since I updated that this morning, and haven't updated rider in a while.

**Comment by shadowmint** [comment]
Still seems to work fine using exactly the same versions of unity and rider on windows, so this appears to be a mac related thing?

**Comment by van800** [comment]
Please try Rider 2018.3 EAP. it is almost release quality.

**Comment by shadowmint** [comment]
Thank you! I can confirm this issue is resolved in Rider 2018.3 EAP 9.

If you want to close this issue as being related to rider not the unity plugin, feel free to close this.
